### PR TITLE
fix: Remove obsolete resource typing

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

In PHP>=8.1, LDAP and FTP resources are always typed objects

## TODO

- [x] Wait for https://github.com/nextcloud/server/pull/44607 and rebase
- [x] Update stubs for gd and ftp
- [x] Remove references to gd resource as well

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by kesselb — APPROVED** [review]
Nice :pray:

File: apps/user_ldap/lib/Wizard.php [hunk 1]
```diff
@@ -1214,9 +1209,7 @@ public function cumulativeSearchOnAttribute(array $filters, string $attr, int $d
 					$dnReadCount++;
 					$foundItems = array_merge($foundItems, $newItems);
 					$dnRead[] = $dn;
-				} while (($state === self::LRESULT_PROCESSED_SKIP
-						|| $this->ldap->isResource($entry))
-						&& ($dnReadLimit === 0 || $dnReadCount < $dnReadLimit));
+				} while ($dnReadLimit === 0 || $dnReadCount < $dnReadLimit);
```
Comment (blizzz) [inline apps/user_ldap/lib/Wizard.php:1212]: You are not taking `$state` into account anymore. Is that intended?

File: build/stubs/ldap.php [hunk 2]
```diff
@@ -25,4 +27,200 @@ final class Result
     final class ResultEntry
     {
     }
+
+}
+
+namespace {
+
+    #ifdef HAVE_ORALDAP
```
Comment (blizzz) [inline build/stubs/ldap.php:35]: what's that?

**Review by blizzz — COMMENTED** [review]
One thing to doublecheck

**Review by nickvergessen — APPROVED** [review]
:elephant:

File: apps/user_ldap/lib/Wizard.php (hunk 1 above)
Comment (come-nc) [reply apps/user_ldap/lib/Wizard.php:1212]: The call to `$this->ldap->isResource($entry)` cannot return `false`, by typing, so the `$state` was not taken into account before either. If this is/was a bug, we can change the condition.
What do you think?

File: build/stubs/ldap.php (hunk 2 above)
Comment (come-nc) [reply build/stubs/ldap.php:35]: oracle ldap :-(
At some point I removed support for compiling php-ldap with that but a contribution added it back :upside_down_face:

File: apps/user_ldap/lib/Wizard.php (hunk 1 above)
Comment (blizzz) [reply apps/user_ldap/lib/Wizard.php:1212]: Right. I was a bit puzzled also because the local `$state` var was not removed (NaaS). All good.

**Review by blizzz — APPROVED** [review]

**Pull request merged**